### PR TITLE
Refactor/refactor modules

### PR DESCRIPTION
**Pull request by NicholasLYang**
Large PR. Basically refactored modules, got CRD working for articles, C for sections

### CODE REVIEW

File: src/js/modules/articles/components/ArticleForm.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  handleSubmit = event => {
+    const { title, content, section } = this.state;
+    const { contributors, onSubmit, sections } = this.props;
+    const formData = { title, content, sections, section, contributors };
+    if (this.validateForm(formData)) {
+      onSubmit(formData);
+    }
+  };
+
+  validateForm = ({ title, content, sections, section, contributors }) => {
```
Comment (JasonLin43212) [inline src/js/modules/articles/components/ArticleForm.js:54]: Fix the errors for filling out the text box.

File: src/js/modules/articles/components/ArticlesTable.js [hunk 2]
```diff
@@ -15,56 +15,61 @@ import { connect } from 'react-redux'
...
 import UserChip from '../../users/components/UserChip'
 import { articlesPreviewSelector } from '../selectors'
-import { saveSelectedArticles, deleteSelectedArticles } from '../actions'
+import { contributorsByArticle } from '../../authorships/selectors'
+import { setSelectedArticles, deleteArticles } from '../actions'
 
 const styles = {
   articlesTable: {
-    maxWidth: "1200px",
+    maxWidth: "800px",
```
Comment (JasonLin43212) [inline src/js/modules/articles/components/ArticlesTable.js]: CSS properties should be in alphabetical order right?

File: src/js/modules/articles/components/CreateArticlePage.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      classes,
+      title,
+      content,
+      throwError,
+      clearError
+    } = this.props;
+    return (
+      <Paper
+        className={classes.createArticlePage}
+        zDepth={2}
```
Comment (JasonLin43212) [inline src/js/modules/articles/components/CreateArticlePage.js:49]: what is zDepth?

File: src/js/modules/articles/components/ArticlesTable.js (hunk 2 above)
Comment (squirrelsquirrel78) [reply src/js/modules/articles/components/ArticlesTable.js]: yup

File: src/js/modules/RoutingApp.js [hunk 4]
```diff
@@ -1,29 +1,34 @@
...
-              <Route path="/articles/new" component={ArticleForm} />
-              <Route exact path="/users" component={UsersList} />
-              <Route path="/sections/new" component={SectionForm} />
-            </div>
+            <MainApp>
+              <Switch>
+                <Route exact path="/articles" component={ArticlesTable} />
+                <Route path="/articles/new" component={CreateArticlePage} />
+                <Route exact path="/users" component={UsersList} />
+                <Route path="/sections/new" component={SectionForm} />
```
Comment (squirrelsquirrel78) [inline src/js/modules/RoutingApp.js:27]: It's a little inconsistent. I would expect `/articles/new` and `/sections/new` to be either named CreateArticlePage and CreateSection or ArticleForm and SectionForm.

File: src/js/modules/articles/components/ArticleForm.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+class ArticleForm extends Component {
+
+  constructor(props) {
+    super(props);
+    const { title, content } = this.props;
+    this.state = {
+      title,
+      content,
+      section: 2
```
Comment (squirrelsquirrel78) [inline src/js/modules/articles/components/ArticleForm.js:26]: Not sure what `section: 2` is supposed to do. If it's a default for drafts, maybe this should be made clearer with a constant or a comment.

File: src/js/modules/articles/components/ArticlesList.js [hunk 6]
```diff
@@ -33,7 +33,7 @@ const ArticlesList = ({ classes, articles }) =>
   </div>;
 
 const mapStateToProps = state => ({
-  list: state.forms.list
+  articles: state.forms.list
```
Comment (squirrelsquirrel78) [inline src/js/modules/articles/components/ArticlesList.js]: A little confusing. So articles are stored in forms not articles?

File: src/js/modules/forms/reducer.js [hunk 7]
```diff
@@ -1,39 +1,12 @@
...
-          );
-          const newContributors = contributors.slice(0, contributorIndex).concat(
-            contributors.slice(contributorIndex + 1)
-          );
-          return { ...state, contributors: newContributors };
-        case SAVE_ARTICLE_FORM_DATA:
-          return { ...state, articleForm: action.payload }
-        case CLEAR_ARTICLE_FORM_DATA:
-          return { ...state, articleForm: {}, contributors: [] }
+
```
Comment (squirrelsquirrel78) [inline src/js/modules/forms/reducer.js]: If nothing happens, just return state.

File: src/js/modules/authorships/selectors.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ * Created by nicholas on 8/4/17.
+ */
+import { createSelector } from 'reselect'
+import { articlesSelector } from '../articles/selectors'
+import { usersSelector } from '../users/selectors'
+
+export const authorshipsSelector = state => state.authorships.list
+
+// Returns object with the format articleId:[users]
+export const contributorsByArticle = createSelector(
```
Comment (squirrelsquirrel78) [inline src/js/modules/authorships/selectors.js]: Naming please.

File: src/js/modules/main/components/MainApp.js [hunk 9]
```diff
@@ -56,8 +43,16 @@ class MainApp extends PureComponent
...
   componentDidMount()
   {
-    this.props.fetchArticles();
-    this.props.fetchSections();
+    const {
+      fetchArticles,
+      fetchSections,
+      fetchAuthorships,
+      fetchUsers,
+    } = this.props
```
Comment (squirrelsquirrel78) [inline src/js/modules/main/components/MainApp.js]: semicolon

File: src/js/modules/main/components/MainApp.js [hunk 10]
```diff
@@ -96,8 +89,14 @@ const VisibleMainApp = connect(
       },
       fetchArticles: () => {
         dispatch(fetchArticles())
+      },
+      fetchAuthorships: () => {
+        dispatch(fetchAuthorships())
+      },
+      fetchUsers: () => {
+        dispatch(fetchUsers())
```
Comment (squirrelsquirrel78) [inline src/js/modules/main/components/MainApp.js]: why? `mapDispatchToProps`???

File: src/js/modules/sections/actionTypes.js [hunk 11]
```diff
@@ -1,5 +1,7 @@
 export const FETCH_SECTIONS_REQUESTED = "forms/FETCH_SECTIONS_REQUESTED";
-
 export const FETCH_SECTIONS_SUCCEEDED = 'forms/FETCH_SECTIONS_SUCCEEDED';
+export const FETCH_SECTIONS_FAILED = 'forms/FETCH_SECTIONS_FAILED';
 
-export const FETCH_SECTIONS_FAILED = 'forms/FETCH_SECTIONS_FAILED';
\ No newline at end of file
+export const CREATE_SECTION_REQUESTED = "forms/CREATE_SECTION_REQUESTED";
+export const CREATE_SECTION_SUCCEEDED = 'forms/CREATE_SECTION_SUCCEEDED';
+export const CREATE_SECTION_FAILED = 'forms/CREATE_SECTION_FAILED';
```
Comment (squirrelsquirrel78) [inline src/js/modules/sections/actionTypes.js:7]: Again, the action naming I mentioned earlier. Please consider it.

File: src/js/modules/sections/components/SectionForm.js [hunk 12]
```diff
@@ -5,9 +5,12 @@ import React, { Component } from 'react'
 import Paper from 'material-ui/Paper'
 import TextField from 'material-ui/TextField'
 import RaisedButton from 'material-ui/RaisedButton'
+import SelectField from 'material-ui/SelectField'
+import MenuItem from 'material-ui/MenuItem'
 import injectSheet from 'react-jss'
 import { createSection } from '../actions'
 import { connect } from 'react-redux'
+import { topLevelSectionsSelector } from '../selectors'
```
Comment (squirrelsquirrel78) [inline src/js/modules/sections/components/SectionForm.js]: Just a reminder about one of my first comments: double quotes vs. single quotes.

**Review by squirrelsquirrel78 — CHANGES_REQUESTED** [review]
(1) Some of your naming doesn't follow convention: https://medium.com/@kylpo/redux-best-practices-eef55a20cc72.
(2) Style: react things should be in double quotes (import statements). Other JS things are find in single quotes.

File: src/js/modules/RoutingApp.js (hunk 4 above)
Comment (NicholasLYang) [reply src/js/modules/RoutingApp.js:27]: Yep, will fix in next branch. When I add editing sections

File: src/js/modules/articles/components/ArticlesList.js (hunk 6 above)
Comment (NicholasLYang) [reply src/js/modules/articles/components/ArticlesList.js]: ...oops. ArticleList will be deprecated soon

File: src/js/modules/forms/reducer.js (hunk 7 above)
Comment (NicholasLYang) [reply src/js/modules/forms/reducer.js]: Shouldn't it do that? Switch breaks then returns state

File: src/js/modules/main/components/MainApp.js (hunk 10 above)
Comment (NicholasLYang) [reply src/js/modules/main/components/MainApp.js]: Yeah yeah

File: src/js/modules/forms/reducer.js (hunk 7 above)
Comment (squirrelsquirrel78) [reply src/js/modules/forms/reducer.js]: yea yea

**Pull request merged**